src/lib/dialog/dialog-container.ts

```diff
@@ -60,6 +62,16 @@ export class MdDialogContainer extends BasePortalHost implements OnDestroy {
     }
 
     let attachResult = this._portalHost.attachComponentPortal(portal);
+    let componentElement = attachResult.location.nativeElement;
+
+    // Add a class that we can use for styling the root element.
+    this._renderer.setElementClass(componentElement, 'md-dialog-root', true);
+
+    // Add flexbox styling if the user is using the `md-dialog-content`.
+    if ('querySelector' in componentElement) {
+      this._renderer.setElementClass(componentElement, 'md-dialog-root-flex',
```

Comment:
I'm not sure we should do this. This will set `display: flex` on the host element of the loaded component, which will potentially change the layout of the contents of that element in a way that the user did not intend; we should generally avoid setting any styles to elements Material doesn't own. 

Why not add an element to the dialog-container template for this (and for `md-dialog-root`)?